### ISSUE TITLE
All widgets disappeared upon update today and the debug console won't open. Just me?

### ISSUE THREAD

**Issue by fishcymbal** [felixhageloh/uebersicht#429]
MacOS 10.15.7

**Comment by skonesam** [comment]
Not just you! Same problems.

The commands are running, though; I have one of my widgets also run a simple command to update "AnyBar".

**Comment by fishcymbal** [comment]
Not sure why I can't open the debug console though -- I thought it was available even if there weren't any errors. Maybe this was changed in the upgrade?

**Comment by felixhageloh** [comment]
If you send them to the foreground again, to they keep working?

**Comment by felixhageloh** [comment]
Do you all have interaction disabled by any chance?

**Comment by kafkasl** [comment]
I had interaction disabled. Enabling it, quitting and restarting the app solved the issue.

**Comment by felixhageloh** [comment]
It's a bug when interaction is disabled - it should work fine if it is enabled. Working on a fix!

**Comment by felixhageloh** [comment]
Should be fixed in 1.6.66 (👹) released earlier today

**Comment by kafkasl** [comment]
works for me 👍

**Comment by fishcymbal** [comment]
The debug console works now, but I find 1.6.66 to still be unstable. When I first upgraded, all my widgets disappeared again, but I was able to jump-start them by setting one of the widgets to "show on all screens" and then setting it back to the main display again. Later, when one of my widgets threw an error while processing, it seems to have triggered all widgets to fail again. The same fix (make one widget show on all screens) got them running again. Now that there's a pattern, I will capture some screenshot video to show it in action next time. Thanks for the active development!

**Comment by felixhageloh** [comment]
@fishcymbal are you still seeing those issues?

**Comment by fishcymbal** [comment]
@felixhageloh Yes I am, but if I'm the only one, then it may be something environmental for me or something with my particular widget settings (I keep 21 of them running at a time). All my widgets keep disappearing throughout the day until I go to the menu and refresh them all. Then they reappear. I can't figure out what the trigger is. --That is, all the widgets _except for one_ that plays a small .gif file on a loop. That one remains open, but the rest disappear. When I refresh, they all come back for a while.

I tried downloading and installing an older version of Üebersicht to see if that would help, but when I checked to see if it was successful, the version number remained unchanged and I was still on the latest version.

I find that I'm now unable to _uncheck_ the option for the widgets to be in the background -- did I miss something about a line of code that is now necessary to enable/disable background status? I only ask in case that's related. 

I don't want to hold things up if it's just me. I just can't figure out the issue.

**Comment by seanrabey** [comment]
Hello,
I am having the same issue with my single widget disappearing throughout the day. I'm running on a MacBook Pro with a second screen. The second screen is the "main" screen and the widget is running on the MacBook pro screen. I have dock hiding turned on, if I reveal the dock in the MacBook pro screen, the widget instantly disappears and I have to hit "Refresh Widgets" for it to come back:
https://youtu.be/XwO86eIzreQ

Let me know if I can provide any other info to help debug this.
Thanks very much!

**Comment by fishcymbal** [comment]
@seanrabey I'm running a MBP with a second screen as well. I'm not using the dock hiding feature, but the behavior you describe sounds similar to what I've seen. Seems like some kind of extended screen/finder refresh/screen refresh issue for both of us.  What kind of widget is disappearing for you? Mine were anything running a shell command. The only one that persisted without issue was something with a simple gif running in a loop.

**Comment by seanrabey** [comment]
@fishcymbal, interesting, my widget is running a bunch of shell commands for display as well.  It grabs the date and time, and some specs about my battery and temperatures for display.  Here is the shell command, it runs every second:

`command: "printf \"%.0f&#37;&nbsp;\" $(/usr/local/bin/istats battery charge --value-only) && printf \"%.0fC<br>\" $(/usr/local/bin/istats cpu temp --value-only) && date +\"%Y-%m-%d&nbsp;<br>%a %b %d&nbsp;<br><span id='mainDate'>%l:%M:%S %p</span>&nbsp;<br>%H:%M:%S %Z<br>\" && date -u +\"%H:%M:%S UTC<br>\""`

**Comment by seanrabey** [comment]
PS: if I "Enable Interaction" it stays, but then it doesn't clear when I use mission control show desktop. A small annoyance, but would be nice :)

**Comment by fishcymbal** [comment]
@seanrabey Wow, that's nothing compared to all the commands I was running and it still fails for both of us. I had a ton of widgets calling scripts and grabbing text files, parsing them out with CoffeeScript and JavaScript, etc.  So it doesn't seem like a matter of complexity, just that they're shell commands. I thought maybe I was trying to do too much. Interesting....

Also notable that the *interaction* function is a triggering issue. That's when my problems started as well, version 6.6.5 I think. I have interaction turned off and have never even looked into how to use it. Wonder if that would be a fix for me as well. 

Sorry, I guess I'm being dense -- I don't follow what you're saying about mission control -- what doesn't clear?

**Comment by seanrabey** [comment]
@fishcymbal seems like it's definitely something to do with the shell commands.
Yes, by enabling interactions, it no longer disappears.
I like to use the mac missions control keyboard shortcut to "show desktop" and when the widget does not have interaction enabled and it's moved to the background, it moves out of the way when I use the "show desktop" keyboard shortcut. But with interaction enabled, it stays stuck in place and doesn't move off to the side like all the other app windows do :)

**Comment by felixhageloh** [comment]
did you both check
- the debug console for any errors?
- the output of the widget command to see if it returns an error?

The fact that you mention that not all widgets have the same issue rules out an issue with the Übersicht window in general

**Comment by fishcymbal** [comment]
Hi @felixhageloh 

Yes, I've checked my debug console many times. I've turned off the widgets that cause errors and the remaining ones still disappear. And I understand your point that the inconsistent behavior might imply that there's no issue with the core program, but on the other hand these widgets persisted without an issue before 1.6.5 so it seems like something about the core functionality has led to the new behavior. 

Don't get me wrong -- it's not a complaint and I appreciate the free software without an expectation of free support. Just trying to help troubleshoot.

**Comment by fishcymbal** [comment]
Update: I finally caught them in the act of disappearing -- They had been running on the desktop for a couple minutes when I opened a processor/memory intensive program (in this case, Adobe Premiere Pro). Before the app's window could obscure the desktop, all but that one looping .gif disappeared. I have to refresh if I want the rest to come back, but I don't see any errors in the log.
